core/modules/file/file.file.inc

```diff
@@ -0,0 +1,317 @@
+ */
+function file_file_update($file) {
+  // Ensure field data is saved since file_save() does not in Drupal 7.
+  field_attach_update('file', $file);
+
+  // Save file metadata.
+  db_delete('file_metadata')->condition('fid', $file->fid)->execute();
+  if (!empty($file->metadata)) {
+    $query = db_insert('file_metadata')->fields(array('fid', 'name', 'value'));
+    foreach ($file->metadata as $name => $value) {
+      $query->values(array(
+        'fid' => $file->fid,
+        'name' => $name,
+        'value' => serialize($value),
+      ));
+    }
+    $query->execute();
+  }
+
+  if (module_exists('image') && file_get_mimetype_type($file) == 'image' && $file->filesize && isset($file->original)) {
+    if (!isset($file->metadata)) {
+      $file->metadata = array();
+    }
+
+    if (!isset($file->original->metadata)) {
+      if (!is_object($file->original)) {
+        $file->original = new stdClass();
+      }
+      $file->original->metadata = array();
+    }
+
+    // If the file has changed dimensions or a new file has been uploaded,
+    // update any image field reference to this file and flush image style
+    // derivatives.
+    $file->metadata += array('width' => NULL, 'height' => NULL);
+    $file->original->metadata += array('width' => NULL, 'height' => NULL);
+    if ($file->filesize != $file->original->filesize || $file->metadata['width'] != $file->original->metadata['width'] || $file->metadata['height'] != $file->original->metadata['height']) {
+      _file_update_image_field_dimensions($file);
+    }
+
+    // Flush image style derivatives whenever an image is updated.
+    image_path_flush($file->uri);
+  }
+
+  // Clear any related field caches.
+  file_invalidate_field_caches($file);
+}
+
+/**
+ * Implements hook_file_delete().
+ */
+function file_file_delete($file) {
+  field_attach_delete('file', $file);
+
+  // This is safe to call since the file's records from the usage table have
+  // not yet been deleted.
+  file_invalidate_field_caches($file);
+
+  // Remove file metadata.
+  db_delete('file_metadata')->condition('fid', $file->fid)->execute();
+
+  // Remove this file from the search index if needed.
+  // This code is implemented in file entity module rather than in search
+  // module because file entity is implementing search module's API, not the
+  // other way around.
+  if (module_exists('search')) {
+    search_reindex($file->fid, 'file');
+  }
+}
+
+/**
+ * Implements hook_file_mimetype_mapping_alter().
+ */
+function file_file_mimetype_mapping_alter(&$mapping) {
+  // Add support for mka and mkv.
+  // @todo Remove when http://drupal.org/node/1443070 is fixed in core.
+  $new_mappings['mka'] = 'audio/x-matroska';
+  $new_mappings['mkv'] = 'video/x-matroska';
+
+  // Add support for weba, webm, and webp.
+  // @todo Remove when http://drupal.org/node/1443070 is fixed in core.
+  $new_mappings['weba'] = 'audio/webm';
+  $new_mappings['webm'] = 'video/webm';
+  $new_mappings['webp'] = 'image/webp';
```

Comment:
Looks like these mappings can be deleted. See d3b960e15895c5f448cb3c1b4c5d4981ab69976f.